plugin/src/utils/run-steps.ts

```diff
@@ -1,15 +1,28 @@
+import { GatsbyNodeApiHelpers } from "./gatsby-types"
+import { IPluginOptions } from "~/models/gatsby-api"
 import { formatLogMessage } from "~/utils/format-log-message"
 import { invokeAndCleanupLeftoverPreviewCallbacks } from "../steps/preview/cleanup"
 import { CODES } from "./report"
 
-const runSteps = async (steps, helpers, pluginOptions, apiName) => {
+type Step = (
+  helpers: GatsbyNodeApiHelpers,
+  pluginOptions: IPluginOptions
+) => Promise<void>
+
+const runSteps = async (
+  steps: Step[],
+  helpers: GatsbyNodeApiHelpers,
+  pluginOptions: IPluginOptions,
+  apiName: string
+): Promise<void> => {
   for (const step of steps) {
     try {
       const { timeBuildSteps } = pluginOptions?.debug ?? {}
       const timeStep =
         typeof timeBuildSteps === `boolean`
           ? timeBuildSteps
-          : timeBuildSteps?.includes(step.name) ||
+          : // TODO: type for array option? or deprecated setting?
+            timeBuildSteps?.includes(step.name) ||
```

Comment:
This was meant to be an internal API for debugging the source plugin really. It allows timing all build steps by setting it true or timing a specific set of steps by including their names as an array.

Reply:
ahh ok! well currently in vscode it says that `pluginOptions.debug.timeBuildSteps` does not exist. `IPluginOptions` doesn't have this field, should I add it?

Reply:
@TylerBarnes ah i see the issue here! so `debug.timeBuildSteps` is listed as a `boolean` in `IPluginOptions`. it should be an array of strings instead?

Reply:
https://github.com/gatsbyjs/gatsby-source-wordpress-experimental/pull/332/commits/cdd8d75c56e9cf71c6fc472e6860c3bcea219c1a

Reply:
It can be either an array of strings or a boolean

Reply:
can it just be an empty array? i guess that would be a breaking change

Reply:
ohhhh i see